### PR TITLE
test(Dropdown): add tests, documentation, and cleanup

### PR DESCRIPTION
**Pull request by saurabhdaware**
Add tests, storybook documentation, and jsdoc.

**Storybook:** https://61c19ee8d3d282003ac1d81c-brgjegznoq.chromatic.com/?path=/docs/components-dropdown-with-select--with-single-select&globals=measureEnabled:false

**JSDoc changes:**
image

### CODE REVIEW

File: packages/blade/src/components/Input/SelectInput/SelectInput.tsx [hunk 1]
```diff
@@ -85,7 +83,7 @@ const _SelectInput = (
         {...baseInputProps}
         as="button"
         componentName={MetaConstants.SelectInput}
-        ref={triggererRef as React.MutableRefObject<HTMLInputElement>}
+        ref={!isReactNative() ? (triggererRef as React.MutableRefObject<HTMLInputElement>) : null}
```
Comment (saurabhdaware) [inline packages/blade/src/components/Input/SelectInput/SelectInput.tsx:86]: React Native Testing library doesn't support refs https://github.com/callstack/react-native-testing-library/issues/1069#issuecomment-1221105858

There are some workarounds but we're anyway not using this ref in React native so removed it from here

File: packages/blade/src/components/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -9,12 +9,17 @@ import { isValidAllowedChildren } from '~utils';
 type DropdownProps = {
   selectionType?: 'single' | 'multiple';
   children: React.ReactNode[];
-};
+}; //
```
Comment (anuraghazra) [inline packages/blade/src/components/Dropdown/Dropdown.tsx:12]:
```suggestion
};
```

File: packages/blade/src/components/Dropdown/DropdownOverlay.native.tsx [hunk 3]
```diff
@@ -8,42 +8,44 @@ import Box from '~components/Box';
...
 
   return (
     <Box position="relative">
-      <StyledClosableArea
+      <StyledCloseableArea
         display={isOpen ? 'flex' : 'none'}
         onPress={() => {
           setIsOpen(false);
         }}
+        testID="closeable-area"
```
Comment (anuraghazra) [inline packages/blade/src/components/Dropdown/DropdownOverlay.native.tsx:37]: possible to use `testID()` function from utils? 

Plus is the data-testid set on web side?

Comment (saurabhdaware) [reply packages/blade/src/components/Dropdown/DropdownOverlay.native.tsx:37]: nope. On web I can use `getByRole` with a name option on listbox/menu but in the `@testing-library/react-native` version that we're using, it doesn't seem to be supported.

Comment (saurabhdaware) [reply packages/blade/src/components/Dropdown/DropdownOverlay.native.tsx:37]: Do I use `testID` util if I only want it in react-native? don't see any harm in adding `data-testid` as well just that I wouldn't recommend using that for web tests 🤔

File: packages/blade/src/components/Dropdown/__tests__/Dropdown.web.test.tsx [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+    // Move to first item
+    await user.keyboard('{ArrowDown}');
+    expect(getActiveDescendant()).toBe('Apple');
+
+    // Move to second item
+    await user.keyboard('{ArrowDown}');
+    expect(getActiveDescendant()).toBe('Mango');
+
+    // Select option
+    await user.keyboard('[Space]');
```
Comment (anuraghazra) [inline packages/blade/src/components/Dropdown/__tests__/Dropdown.web.test.tsx:182]: Let's also test PageUp/PageDown, moves focus to first/last element

File: packages/blade/src/components/Dropdown/__tests__/Dropdown.web.test.tsx [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+      </Dropdown>,
+    );
+
+    const selectInput = getByRole('combobox', { name: 'Fruits' });
+
+    expect(selectInput).toBeInTheDocument();
+    expect(selectInput.textContent).toBe('Select Option');
+    expect(container.querySelector('[role=listbox]')).not.toBeVisible();
+
+    const getActiveDescendant = (): string | null | undefined => {
```
Comment (anuraghazra) [inline packages/blade/src/components/Dropdown/__tests__/Dropdown.web.test.tsx:207]: Move this outside maybe? We are redeclaring again.

File: packages/blade/src/components/Dropdown/__tests__/Dropdown.web.test.tsx [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+
+    expect(selectInput.textContent).toBe('Mango');
+
+    // Ensure menu did not close
+    expect(getByRole('listbox', { name: 'Fruits' })).toBeVisible();
+
+    // Move to third item and select
+    await user.keyboard('{ArrowDown}');
+    expect(getActiveDescendant()).toBe('Orange');
+    await user.keyboard('[Space]');
```
[the remark this answers is not in the document]
Comment (saurabhdaware) [reply packages/blade/src/components/Dropdown/__tests__/Dropdown.web.test.tsx:235]: `{` is for `event.key` and `[` is for `event.code`. 

In code we're using `event.key` so I kept that similar here.

For spacebar, `event.key` is ` ` (empty space) so here it becomes `{ }` which looks a bit strange 😅 so using `[Space]` for spacebar

**Review by anuraghazra — APPROVED** [review]
Mostly lgtm.

File: packages/blade/src/components/Dropdown/__tests__/Dropdown.web.test.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (anuraghazra) [reply packages/blade/src/components/Dropdown/__tests__/Dropdown.web.test.tsx:235]: Got it, yeah I used [] only on other test, got confused with this {}

File: packages/blade/src/components/Dropdown/DropdownOverlay.native.tsx (hunk 3 above)
Comment (anuraghazra) [reply packages/blade/src/components/Dropdown/DropdownOverlay.native.tsx:37]: > Do I use testID util if I only want it in react-native? don't see any harm in adding data-testid as well just that I wouldn't recommend using that for web tests 🤔

Sure, if we are not doing any testing with the testId let's not add it on the web.

File: packages/blade/src/components/Dropdown/__tests__/Dropdown.web.test.tsx (hunk 5 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Dropdown/__tests__/Dropdown.web.test.tsx]: done

File: packages/blade/src/components/Dropdown/__tests__/Dropdown.web.test.tsx (hunk 4 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Dropdown/__tests__/Dropdown.web.test.tsx:213]: Added cases for Home, End, PageUp, and PageDown in focus move test 👯

**Pull request merged**